install.sh

```diff
@@ -361,24 +348,10 @@ echo "  ___ _                               _
  \___/|   |_/   |_/\__/  |  |  |_/|__/\__/   |_/|__/  \_/ \_/
 "
 
-if [[ "${CREW_PREFIX}" != "/usr/local" ]]; then
-  echo_info "\n$
-Since you have installed Chromebrew in a directory other than '/usr/local',
-you need to run these commands to complete your installation:
-"
-
-  echo_intra "
-echo 'export CREW_PREFIX=${CREW_PREFIX}' >> ~/.bashrc
-echo 'export PATH=\"\${CREW_PREFIX}/bin:\${CREW_PREFIX}/sbin:\${PATH}\"' >> ~/.bashrc
-echo 'export LD_LIBRARY_PATH=${CREW_PREFIX}/lib${LIB_SUFFIX}' >> ~/.bashrc
-source ~/.bashrc"
```

Comment:
We will still need the `source ~/.bashrc` message to get all the `/usr/local/etc/..` profile stuff, right?

Reply:
I dont know exactly what you mean. There is no need to source for the path variables, as they are still set in the current runtime and will be set on later runtimes, but I get the feeling you arent talking about those.

Reply:
Still need a `echo 'source ~/.bashrc'` message, otherwise, none of the profile environment is set initially.

Reply:
`echo 'source ~/.bashrc'` not `source ~/.bashrc`?

Reply:
Yes, the user need to executes that in their shell session right after the install completes, otherwise, they need to logout and login again to get the profile environment set up.

Reply:
Maybe a message with both options would be the most informative?

Reply:
Can we just end the script with `exec bash`?  That way users end up in a subshell which will involve `~/.bashrc`...

Reply:
The message can then just suggest rebooting when convenient...

Alternately one could do a `killall bash` at the end of the script which will drop people back to the `crosh` prompt and force people to run `shell` again....

Reply:
`exec bash` should work, dropping people back to `crosh` may be a little confusing for them.

Reply:
Note: `exec bash` will not stop the parent shell (the shell which calls `curl ... | bash`), it will only replace the subshell (the `| bash` in the above). As a result, the parent shell will always run in the background (and `crosh` will crash when we use `kill -9` to kill it)

Also, we need to redirect stdin (that connects the `curl` pipe) back to `/dev/tty` before `exec bash`, otherwise it will not work

The only way to replace both the parent shell and subshell is `curl ... | exec bash`

Reply:
It's nice to have the environment since all the nice things like sommelier can get setup easily?

Speaking of which, it might be nice to ask the user if they want graphical apps to work after the install, and if so install sommelier, and then do the exec bash, as that will then immediately give them a working sommelier setup.

Reply:
Let's revisit the sommelier stuff later, I've got a couple thoughts about how we handle it atm anyways.

So, @supechicken, what exactly should I put?

i.e. what is the last line of the script (the one that changes the shell) and what is the command after `curl ... |`?

Also:
As I understand it we are in a subshell during the installation, so would killing that drop people back to crosh or back to the original shell?

Would the original shell have the variables set up?

Reply:
Sorry about missing this message😅

To replace the parent shell, we have two ways:
- This replaces the current running shell with our installation shell and drops us to an interactive bash shell after installation is done (requires no changes to `install.sh`)
```shell
$ exec bash --init-file <(curl -Ls git.io/vddgY)
```

- This runs `install.sh` directly in the existing shell, returning to the original shell once installed (requires no changes to `install.sh`)
```shell
$ curl -Ls git.io/vddgY | source /proc/self/fd/0
```

Be aware that both methods will pass ALL variables we set in `install.sh` to the interactive shell, so we will need to unset those variables before giving back control to the user with `unset`.

Reply:
Better late than never :p

Anyways, as we stay in the same shell with your first solution, we can take user input! Two birds with one stone!

Reply:
This still doesn't resolve the interactive shell issue with environment variables.  We need to execute `source ~/.bashrc` and the user needs to run that from their shell session.

Reply:
What do we need to `source ~/.bashrc` for? We have all the variables we need in terms of paths and library paths in the shell, as they've been passed from the installer. Those variables will also be in the shell on the next startup, because we put them in the crew profile directories.

Reply:
sourcing ~/.bashrc pulls in all the env.d and bash.d files via the `source /usr/local/etc/profile` line...

Reply:
This is absolutely necessary for the `aarch64` installs, without which LD_LIBRARY_PATH will be wrong and the installs will not work properly.

Reply:
We already have the relevant variables from those files in this shell, though. And the next shell the user opens will have those variables, no?

Reply:
While looking more closely it turns out the line in the install script that prompts you to `source ~/.bashrc` is still there, so...